src/Compiler/CompilerUtils.cpp

```diff
@@ -208,6 +216,25 @@ string getToolPath(string tool) {
     return string();
 }
 
+//
+// Return error message for given error number for opening file at given path
+// If error number is not given, investigate it by opening it.
+//
+string getErrorMessageforFileOpeningErrors(string path, int _errno = -1,
+    int _flags = -1, int _mode = -1) {
+  // If errno not given investigate the error by opening the path
+  if (_errno < 0) {
+    int flags = (_flags >= 0) ? _flags : (O_CREAT | O_WRONLY);
+    int mode = (_mode >= 0) ? _mode : (S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+    fd = open(path, flags, mode);
+    _errno = errno;
+    close(fd);
+  }
+  char dir[PATH_SIZE];
+  getcwd(dir, PATH_SIZE);
+  string msg = strerror(_errno) + "(" + _errno + ") for " + path + " at " dir;
+  return errorMessage;
```

Comment:
I do not 'see' the declaration of `errorMessage`, suggest:

```
return  strerror(_errno) + "(" + _errno + ") for " + path + " at " + dir;
```

Reply:
fixed.